src/WholeBodyControllers/src/QPInverseKinematics.cpp

```diff
@@ -555,28 +560,55 @@ void WalkingQPIK::evaluateGradientVector()
            * (regularizationTerm - jointPosition);
     else
     {
-        auto leftHandCorrectionLinear(iDynTree::toEigen(m_leftHandCorrection.getLinearVec3()));
-        auto leftHandCorrectionAngular(iDynTree::toEigen(m_leftHandCorrection.getAngularVec3()));
-        auto rightHandCorrectionLinear(iDynTree::toEigen(m_rightHandCorrection.getLinearVec3()));
-        auto rightHandCorrectionAngular(iDynTree::toEigen(m_rightHandCorrection.getAngularVec3()));
+        Eigen::Map<Eigen::Vector3d> leftHandCorrectionLinear(iDynTree::toEigen(m_leftHandCorrection.getLinearVec3()));
+        Eigen::Map<Eigen::Vector3d> leftHandCorrectionAngular(iDynTree::toEigen(m_leftHandCorrection.getAngularVec3()));
+        Eigen::Map<Eigen::Vector3d> rightHandCorrectionLinear(iDynTree::toEigen(m_rightHandCorrection.getLinearVec3()));
+        Eigen::Map<Eigen::Vector3d> rightHandCorrectionAngular(iDynTree::toEigen(m_rightHandCorrection.getAngularVec3()));
+
+        auto saturationLambda = [](const Eigen::Map<Eigen::Vector3d>& input, double saturationValue) -> Eigen::Vector3d {
+            if (saturationValue > 0)
+            {
+                if (saturationValue < 1e-10)
+                {
+                    Eigen::Vector3d nullVector;
+                    nullVector.setZero();
+
+                    return nullVector;
```

Comment:
Why is it required?

Reply:
In the other case below, the saturation value is at the denominator. I would have liked to avoid infinite values there.